### PR TITLE
modified rhtap installation to accept the installation options generated by pict

### PR DESCRIPTION
**Pull request by xinredhat**
modified rhtap installation to accept the installation options generated by pict

### CODE REVIEW

File: integration-tests/pict-models/default.pict [hunk 1]
```diff
@@ -2,9 +2,16 @@
 # This is file used for integration tests (PR checks, post-merge checks)
 #
 
+# OCP: 4.17, 4.16
+# ACS: valid values:  new, hosted
+# Registry: valid values:  quay, quayio, artifact
```
Comment (prietyc123) [inline integration-tests/pict-models/default.pict]:
```suggestion
# Registry: valid values:  quay, quay.io, artifactory
```

**Review by prietyc123 — CHANGES_REQUESTED** [review]

**Comment by xinredhat** [comment]
The both sub-pipelineruns of [pr-e2e-tests-5s68f](https://console.redhat.com/application-pipeline/workspaces/rhtap-shared-team/applications/rhtap-cli/pipelineruns/pr-e2e-tests-5s68f) Succeeded, I think we can merge this PR. @prietyc123 WDYT?

[e2e-4.17-hkfkh](https://console.redhat.com/application-pipeline/workspaces/rhtap-shared-team/applications/rhtap-cli/pipelineruns/e2e-4.17-hkfkh)
[e2e-4.16-x6w6g](https://console.redhat.com/application-pipeline/workspaces/rhtap-shared-team/applications/rhtap-cli/pipelineruns/e2e-4.16-x6w6g)

File: integration-tests/pipelines/e2e-main-pipeline.yaml [hunk 2]
```diff
@@ -117,18 +113,30 @@ spec:
...
+                  --param tpa_config="$TPA" \
+                  --param scm_config="$SCM" \
+                  --param pipeline_config="$PIPELINE" \
                   --use-param-defaults \
                   --labels "appstudio.openshift.io/component=${KONFLUX_COMPONENT_NAME}" \
                   --labels "appstudio.openshift.io/application=${KONFLUX_APPLICATION_NAME}" \
                   --labels "pipelines.appstudio.openshift.io/type=$(context.pipelineRun.name)" \
                   --labels "test.appstudio.openshift.io/scenario=pr-e2e-tests" \
-                  --prefix-name "e2e-$version"\
+                  --prefix-name "e2e-$OCP_VERSION"\
```
Comment (prietyc123) [inline integration-tests/pipelines/e2e-main-pipeline.yaml:141]: If there is multiple configuration gets generated with just say two configuration for 4.16, How this plr will be differentiated?

File: integration-tests/pipelines/rhtap-cli-e2e.yaml [hunk 3]
```diff
@@ -38,6 +38,21 @@ spec:
...
       type: string
+    - name: acs_config
+      type: string
+      description: "The ACS option for rhtap-cli installation. Valid values are 'new' and 'hosted'."
+    - name: tpa_config
+      type: string
+      description: "The TPA option for rhtap-cli installation. Valid values are 'new' and 'hosted'."
+    - name: registry_config
+      type: string
+      description: "The image registry option for rhtap-cli installation. Valid values are 'quay', 'quayio' and 'artifactory'."
```
Comment (prietyc123) [inline integration-tests/pipelines/rhtap-cli-e2e.yaml]:
```suggestion
      description: "The image registry option for rhtap-cli installation. Valid values are 'quay', 'quay.io' and 'artifactory'."
```

File: integration-tests/pipelines/rhtap-cli-e2e.yaml [hunk 4]
```diff
@@ -38,6 +38,21 @@ spec:
     - name: konflux-test-infra-secret
       description: The name of secret where testing infrastructures credentials are stored.
       type: string
+    - name: acs_config
+      type: string
+      description: "The ACS option for rhtap-cli installation. Valid values are 'new' and 'hosted'."
+    - name: tpa_config
+      type: string
+      description: "The TPA option for rhtap-cli installation. Valid values are 'new' and 'hosted'."
+    - name: registry_config
```
Comment (prietyc123) [inline integration-tests/pipelines/rhtap-cli-e2e.yaml:47]: Should we consider any default value? Same question for other params as well

File: integration-tests/scripts/install.sh [hunk 5]
```diff
@@ -9,19 +9,17 @@ if [ -f "$HOME/rhtap-cli-ci-kubeconfig" ]; then
...
-bitbucket_enabled="${bitbucket_enabled:-false}"
-
-echo "[INFO] acs_install_enabled=$acs_install_enabled"
-echo "[INFO] quay_install_enabled=$quay_install_enabled"
-echo "[INFO] github_enabled=$github_enabled"
-echo "[INFO] gitlab_enabled=$gitlab_enabled"
-echo "[INFO] jenkins_enabled=$jenkins_enabled"
+acs_config="${acs_config:-new}" # new, hosted
+tpa_config="${tpa_config:-new}" # new, hosted
+registry_config="${registry_config:-quay}" # quay, quayio, artifactory
```
Comment (prietyc123) [inline integration-tests/scripts/install.sh:14]:
```suggestion
registry_config="${registry_config:-quay}" # quay, quay.io, artifactory
```

File: integration-tests/scripts/install.sh [hunk 6]
```diff
@@ -105,17 +109,17 @@ jenkins_integration() {
...
 
-    ./bin/rhtap-cli integration --kube-config "$KUBECONFIG" gitlab --token "${GITLAB__TOKEN}"
+    ./bin/rhtap-cli integration --kube-config "$KUBECONFIG" gitlab --token "${GITLAB__TOKEN}" 
   fi
 }
 
-quay_integration() {
-  if [[ "${quay_install_enabled}" == "false" ]]; then
+quayio_integration() {
+  if [[ "${registry_config}" == "quayio" ]]; then
```
Comment (prietyc123) [inline integration-tests/scripts/install.sh]:
```suggestion
  if [[ "${registry_config}" == "quay" ]]; then
```
or 
```suggestion
  if [[ "${registry_config}" == "quay.io" ]]; then
```

File: integration-tests/pipelines/e2e-main-pipeline.yaml (hunk 2 above)
Comment (xinredhat) [reply integration-tests/pipelines/e2e-main-pipeline.yaml:141]: ah, good catch

File: integration-tests/tasks/rhtap-install.yaml [hunk 7]
```diff
@@ -64,6 +54,16 @@ spec:
           exit 1
         fi
 
+        # Generate the .env file for rhtap-cli installation based on the configuration given in the pict file
+        if [ -f .env ]; then
+            rm .env
+        fi
+        echo "export acs_config=\"$(params.acs_config)\"" > .env
```
Comment (prietyc123) [inline integration-tests/tasks/rhtap-install.yaml]: Do you mean 
```suggestion
        echo "export acs_config=\"$(params.acs_config)\"" >> .env
```

**Review by prietyc123 — CHANGES_REQUESTED** [review]

**Comment by prietyc123** [comment]
> The both sub-pipelineruns of [pr-e2e-tests-5s68f](https://console.redhat.com/application-pipeline/workspaces/rhtap-shared-team/applications/rhtap-cli/pipelineruns/pr-e2e-tests-5s68f) Succeeded, I think we can merge this PR. @prietyc123 WDYT?
> 
> [e2e-4.17-hkfkh](https://console.redhat.com/application-pipeline/workspaces/rhtap-shared-team/applications/rhtap-cli/pipelineruns/e2e-4.17-hkfkh) [e2e-4.16-x6w6g](https://console.redhat.com/application-pipeline/workspaces/rhtap-shared-team/applications/rhtap-cli/pipelineruns/e2e-4.16-x6w6g)

makes sense merging it. However I have added few small comments. Could you please take a look

File: integration-tests/pipelines/e2e-main-pipeline.yaml (hunk 2 above)
Comment (xinredhat) [reply integration-tests/pipelines/e2e-main-pipeline.yaml:141]: This is  prefix-name, the pipeline name will append a random string, for example `e2e-4.16-rkx6m`.

File: integration-tests/pipelines/rhtap-cli-e2e.yaml (hunk 4 above)
Comment (xinredhat) [reply integration-tests/pipelines/rhtap-cli-e2e.yaml:47]: I would like to add a `default.pict` in repository to specify the default values. For example [PR#203](https://github.com/redhat-appstudio/rhtap-e2e/pull/203/files)  So the modification of install.sh won't break the expected result. If we need to  consider any default value in task or pipline  yaml files in the future we can set default values.

**Pull request merged**